boards/msp_exp432p401r/src/main.rs

```diff
@@ -0,0 +1,163 @@
+#![no_std]
+#![cfg_attr(not(doc), no_main)]
+#![feature(asm, core_intrinsics)]
+// #![deny(missing_docs)]
+
+use kernel::capabilities;
+use kernel::common::dynamic_deferred_call::DynamicDeferredCall;
+use kernel::common::dynamic_deferred_call::DynamicDeferredCallClientState;
+use kernel::component::Component;
+use kernel::hil::watchdog::Watchdog;
+use kernel::Platform;
+use kernel::{create_capability, debug, static_init};
+
+pub mod io;
+
+// Number of concurrent processes this platform supports.
+const NUM_PROCS: usize = 4;
+
+// Actual memory for holding the active process structures.
+static mut PROCESSES: [Option<&'static dyn kernel::procs::ProcessType>; NUM_PROCS] =
+    [None; NUM_PROCS];
+
+// Static reference to chip for panic dumps.
+static mut CHIP: Option<&'static msp432::chip::Msp432> = None;
+
+// How should the kernel respond when a process faults.
+const FAULT_RESPONSE: kernel::procs::FaultResponse = kernel::procs::FaultResponse::Panic;
+
+// RAM to be shared by all application processes.
+#[link_section = ".app_memory"]
+static mut APP_MEMORY: [u8; 32768] = [0; 32768];
+
+// Force the emission of the `.apps` segment in the kernel elf image
+// NOTE: This will cause the kernel to overwrite any existing apps when flashed!
+#[used]
+#[link_section = ".app.hack"]
+static APP_HACK: u8 = 0;
+
+/// Dummy buffer that causes the linker to reserve enough space for the stack.
+#[no_mangle]
+#[link_section = ".stack_buffer"]
+pub static mut STACK_MEMORY: [u8; 0x1000] = [0; 0x1000];
+
+/// A structure representing this platform that holds references to all
+/// capsules for this platform.
+struct MspExp432P401R {
+    led: &'static capsules::led::LED<'static, msp432::gpio::Pin>,
+    console: &'static capsules::console::Console<'static>,
+}
+
+/// Mapping of integer syscalls to objects that implement syscalls.
+impl Platform for MspExp432P401R {
+    fn with_driver<F, R>(&self, driver_num: usize, f: F) -> R
+    where
+        F: FnOnce(Option<&dyn kernel::Driver>) -> R,
+    {
+        match driver_num {
+            capsules::led::DRIVER_NUM => f(Some(self.led)),
+            capsules::console::DRIVER_NUM => f(Some(self.console)),
+            _ => f(None),
+        }
+    }
+}
+
+#[no_mangle]
+pub unsafe fn reset_handler() {
+    msp432::init();
+    msp432::wdt::WATCHDOG.stop();
+    msp432::sysctl::SYSCTL.enable_all_sram_banks();
+    msp432::pcm::PCM.set_high_power();
+    msp432::flctl::FLCTL.set_waitstates(msp432::flctl::WaitStates::_1);
+    msp432::flctl::FLCTL.set_buffering(true);
+
+    // Setup 48MHz from external oscillator
+    msp432::gpio::PINS[msp432::gpio::PinNr::PJ_2 as usize].enable_primary_function();
+    msp432::gpio::PINS[msp432::gpio::PinNr::PJ_3 as usize].enable_primary_function();
+    msp432::cs::CS.set_clk_48mhz();
+    msp432::cs::CS.set_smclk_12mhz();
+
```

Comment:
Here would be a good place to setup the debug GPIO with `kernel::debug::assign_gpios`

Reply:
Sure, of course! I totally forgot about the GPIOs besides the LED.